framework/src/markers/ComboMarker.C

```diff
@@ -24,10 +25,31 @@ ComboMarker::validParams()
 }
 
 ComboMarker::ComboMarker(const InputParameters & parameters)
-  : Marker(parameters), _names(parameters.get<std::vector<MarkerName>>("markers"))
+  : Marker(parameters),
+    _names(parameters.get<std::vector<MarkerName>>("markers")),
+    _block_restriction_mismatch(false)
 {
   for (const auto & marker_name : _names)
     _markers.push_back(&getMarkerValue(marker_name));
+
+  for (const auto & marker_name : _names)
+    _marker_variables.push_back(&_fe_problem.getVariable(_tid, marker_name));
+
+  // Check block restrictions
+  std::string other_block_restricted = "";
+  for (const auto & marker_name : _names)
+    if (blockIDs() != _fe_problem.getVariable(_tid, marker_name).blockIDs())
```

Comment:
Can you consolidate the two loops, only call `_subproblem.getVariable` once?

Reply:
done